packages/pods-core/src/builtin/AirtablePod.ts

```diff
@@ -0,0 +1,187 @@
+          [key]: _.get(note, `${value}`).toString(),
+        };
+      }
+      return { fields };
+    });
+    return data;
+  }
+
+  async processNote(opts: AirtableExportPodProcessProps) {
+    const {
+      filteredNotes,
+      apiKey,
+      baseId,
+      tableName,
+      checkpoint,
+      srcFieldMapping,
+    } = opts;
+
+    const headers = {
+      Authorization: `Bearer ${apiKey}`,
+      "Content-Type": "application/json",
+    };
+
+    const records = this.notesToSrcFieldMap(filteredNotes, srcFieldMapping);
+    const chunks = _.chunk(records, 10);
+
+    // rate limiter method
+    const sendRequest = async () => {
+      let total: number = 0;
+      chunks.forEach(async (record) => {
+        await limiter.removeTokens(1);
+        const data = JSON.stringify({ records: record });
+        try {
+          const result = await axios.post(
+            `https://api.airtable.com/v0/${baseId}/${tableName}`,
+            data,
+            { headers: headers }
+          );
+          total = total + result.data.records.length;
+          if (total === filteredNotes.length) {
+            const timestamp = filteredNotes[filteredNotes.length - 1].created;
+            fs.writeFileSync(checkpoint, timestamp.toString(), {
+              encoding: "utf8",
+            });
+          }
+        } catch (error) {
+          console.log("failed to export all the notes. Error: ", error);
+          throw Error("error");
+        }
+      });
+    };
+    sendRequest();
+  }
+
+  verifyDir(dest: URI) {
+    const basePath = path.dirname(dest.fsPath);
+    const checkpoint = path.join(
+      basePath,
+      "pods",
+      ID,
+      "airtable-pod.lastupdate",
+      "checkpoint.txt"
+    );
+    fs.ensureDirSync(path.dirname(checkpoint));
+    return checkpoint;
+  }
+
+  //filters the notes of src hierarchy given from all the notes
+  filterNotes(notes: NoteProps[], srcHierarchy: string) {
+    return notes.filter((note) => note.fname.includes(srcHierarchy));
+  }
+
+  async plant(opts: ExportPodPlantOpts) {
+    const { notes, config, dest } = opts;
+    const { apiKey, baseId, tableName, srcFieldMapping, srcHierarchy } =
+      config as AirtableExportConfig;
+    const checkpoint: string = this.verifyDir(dest);
+
+    let filteredNotes: NoteProps[] =
+      srcHierarchy === "root" ? notes : this.filterNotes(notes, srcHierarchy);
+    filteredNotes = _.orderBy(filteredNotes, ["created"], ["asc"]);
+    if (fs.existsSync(checkpoint)) {
+      const lastUpdatedTimestamp: number = Number(
+        fs.readFileSync(checkpoint, { encoding: "utf8" })
+      );
+      filteredNotes = filteredNotes.filter(
+        (note) => note.created > lastUpdatedTimestamp
+      );
+    }
+
+    if (filteredNotes.length > 0) {
+      await this.processNote({
+        filteredNotes,
+        apiKey,
+        baseId,
+        tableName,
+        srcFieldMapping,
+        srcHierarchy,
+        checkpoint,
+      });
+    } else {
+      throw Error(
```

Comment:
instead of throwing an error, can we change the output of `plant` to return `{notes, status}` were notes is the notes that were updated and status is the following enum?

```ts
enum POD_STATUS {
	NO_NEW_ENTRIES = "no_new_entries",
	OK = "ok",
	ERROR = "ERROR"
}

```

Reply:
if we do not throw error, the normal execution does not break and  ../plugin-core/src/commands/ExportPod.ts has 
```js
window.showInformationMessage(`done exporting. destination: ${dest}`);
```
 which gives msg as done exporting, even if it is not. 
Can we make changes in this file as well to display message as per status?

Reply:
got it. yeah, lets save that for a next task then since we have some bigger structural changes as well to integrate